pkg/syncer/metadatasyncer.go

```diff
@@ -801,6 +847,220 @@ func addLabelsToTopologyVCMap(ctx context.Context, nodeTopoObj csinodetopologyv1
+				log.Errorf("updateStoragePolicyUsage failed. err: %v", err)
+			}
+			log.Infof("cnsvolumeoperationrequestCRUpdated: Successfully increased the reserved field by %v "+
+				"for storagepolicyusage CR: %q", newcnsvolumeoperationrequestObj.Status.StorageQuotaDetails.Reserved.Value(),
+				storagePolicyUsageCR.Name)
+		} else {
+			// This is a case where CSI Driver container decreases the value of "reserved" value in
+			// CnsVolumeOperationRequest after a successful CreateVolume operation. And subsequently,
+			// the "reserved" field in StoragePolicyUsage needs to be decreased based on the CnsVolumeOperationRequest
+			// "reserved" field. Also, the "used" field in StoragePolicyUsage needs to be increased.
+			storagePolicyUsageCR.Status.ResourceTypeLevelQuotaUsage.Reserved.Sub(
+				*resource.NewQuantity(oldcnsvolumeoperationrequestObj.Status.StorageQuotaDetails.Reserved.Value(),
+					oldcnsvolumeoperationrequestObj.Status.StorageQuotaDetails.Reserved.Format))
+			storagePolicyUsageCR.Status.ResourceTypeLevelQuotaUsage.Used.Add(
+				*resource.NewQuantity(oldcnsvolumeoperationrequestObj.Status.StorageQuotaDetails.Reserved.Value(),
+					oldcnsvolumeoperationrequestObj.Status.StorageQuotaDetails.Reserved.Format))
+			err := updateStoragePolicyUsage(ctx, cnsOperatorClient, storagePolicyUsageCR)
+			if err != nil {
+				log.Errorf("updateStoragePolicyUsage failed. err: %v", err)
+			}
+			log.Infof("cnsvolumeoperationrequestCRUpdated: Successfully decreased the reserved field by %v "+
+				"for storagepolicyusage CR: %q", oldcnsvolumeoperationrequestObj.Status.StorageQuotaDetails.Reserved.Value(),
+				storagePolicyUsageCR.Name)
+			log.Infof("cnsvolumeoperationrequestCRUpdated: Successfully increased the used field by %v "+
+				"for storagepolicyusage CR: %q", oldcnsvolumeoperationrequestObj.Status.StorageQuotaDetails.Reserved.Value(),
+				storagePolicyUsageCR.Name)
+		}
+	}
+}
+
+func updateStoragePolicyUsage(ctx context.Context, client client.Client,
+	instance *storagepolicyusagev1alpha1.StoragePolicyUsage) error {
+	log := logger.GetLogger(ctx)
+	err := client.Update(ctx, instance)
+	if err != nil {
+		if apierrors.IsConflict(err) {
+			log.Infof("Observed conflict while updating StoragePolicyUsage instance %q in namespace %q."+
+				"Reapplying changes to the latest instance.", instance.Name, instance.Namespace)
+
+			// Fetch the latest instance version from the API server and apply changes on top of it.
+			latestInstance := &storagepolicyusagev1alpha1.StoragePolicyUsage{}
+			err = client.Get(ctx, k8stypes.NamespacedName{Name: instance.Name, Namespace: instance.Namespace}, latestInstance)
+			if err != nil {
+				log.Errorf("Error reading the StoragePolicyUsage with name: %q on namespace: %q. Err: %+v",
+					instance.Name, instance.Namespace, err)
+				// Error reading the object - return error
+				return err
+			}
+			latestInstance.Status = *instance.Status.DeepCopy()
+
+			err := client.Update(ctx, latestInstance)
```

Comment:
We should use "patch" rather than "update" as "update" causes lots of API update conflicts.

Reply:
Done